src/log.cpp

```diff
@@ -201,7 +203,15 @@ LogStream::~LogStream()
     if (type_ == LogType::DEBUG)
       prefix = "[" + log_file_ + ":" + std::to_string(log_line_) + "]\n";
     sink_.take_input(type_, loc_, out_, prefix + buf_.str());
+    if (type_ == LogType::FATAL)
+      abort();
   }
 }
 
+[[noreturn]] LogStreamFatal::~LogStreamFatal()
+{
+  sink_.take_input(type_, loc_, out_, buf_.str());
+  abort();
```

Comment:
Shouldn't the body of this destructor be empty? The base class destructor already handles all the work, right?

Reply:
Oh that's my bad, the base destructor shouldn't have done this:
```
    if (type_ == LogType::FATAL)
      abort();
```
`~LogStreamFatal` should call `abort` instead. Otherwise the compiler would complain `function declared 'noreturn' should not return [-Winvalid-noreturn]`